src/bigistore.c

```diff
@@ -647,6 +648,185 @@ Datum bigistore_each(PG_FUNCTION_ARGS)
     SRF_RETURN_DONE(funcctx);
 }
 
+/*
+ * Verify function caller can handle a tuplestore result, and set up for that.
+ *
+ * Note: if the caller returns without actually creating a tuplestore, the
+ * executor will treat the function result as an empty set.
+ */
+static void
+prepTuplestoreResult(FunctionCallInfo fcinfo)
```

Comment:
That looks like the code "borrowed" from dblink.c, perhaps we should credit them for it in the file header.
However, I think we can get rid of this function and just inline it (as it is used only once).